cloud: update docs for Azure support (#21088)
pingcap/docs: tidb-cloud/setup-azure-self-hosted-kafka-private-link-service.md

    - Manage virtual machines
    - Manage virtual networks
    - Manage load balancers
    - Manage private link services
    - Connect to virtual machines to configure Kafka nodes

2. [Create a TiDB Cloud Dedicated cluster](/tidb-cloud/create-tidb-cluster.md) on Azure if you do not have one.

3. Get the Kafka deployment information from your [TiDB Cloud Dedicated](/tidb-cloud/select-cluster-tier.md#tidb-cloud-dedicated) cluster.

    1. In the [TiDB Cloud console](https://tidbcloud.com), navigate to the [**Clusters**](https://tidbcloud.com/console/clusters) page, and then click the name of your target cluster to go to its overview page.
    2. In the left navigation pane, click **Changefeed**.
    3. On the **Changefeed** page, click **Create Changefeed** in the upper-right corner.
        1. In **Destination**, select **Kafka**.
        2. In **Connectivity Method**, select **Private Link**.
    4. Note down the region information and the subscription of the TiDB Cloud Azure account in **Reminders before proceeding**. You will use it to authorize TiDB Cloud to access the Kafka Private Link service.

>     5. Enter a unique **Kafka Advertised Listener Pattern** for your Kafka Private Link service.

Author: ```suggestion
    5. Generate the **Kafka Advertised Listener Pattern** for your Kafka Private Link service by providing a unique random string.
```

    process.roles=broker,controller
    node.id=2
    controller.quorum.voters=1@{broker-node-1-ip}:29092,2@{broker-node-2-ip}:29092,3@{broker-node-3-ip}:29092
    listeners=INTERNAL://0.0.0.0:9092,CONTROLLER://0.0.0.0:29092,EXTERNAL://0.0.0.0:39092
    inter.broker.listener.name=INTERNAL
    advertised.listeners=INTERNAL://{broker-node-2-ip}:9092,EXTERNAL://b2.abc.eastus.azure.3199745.tidbcloud.com:9094
    controller.listener.names=CONTROLLER
    listener.security.protocol.map=INTERNAL:PLAINTEXT,CONTROLLER:PLAINTEXT,EXTERNAL:PLAINTEXT,SSL:SSL,SASL_PLAINTEXT:SASL_PLAINTEXT,SASL_SSL:SASL_SSL
    log.dirs=./data
    ```

    ```properties
    # broker-node-3 ~/config/server.properties
    # 1. Replace {broker-node-1-ip}, {broker-node-2-ip}, {broker-node-3-ip} with the actual IP addresses.
    # 2. Configure EXTERNAL in "advertised.listeners" based on the "Kafka Advertised Listener Pattern" in the "Prerequisites" section.
    # 2.1 The pattern is "<broker_id>.abc.eastus.azure.3199745.tidbcloud.com:<port>".

>     # 2.2 So the EXTERNAL can be "abc.eastus.azure.3199745.tidbcloud.com:9095". Replace <broker_id> with "b" prefix plus "node.id" properties, and replace <port> with a unique port (9095) in the port range of the EXTERNAL advertised listener.

Author: From Gemini: https://github.com/pingcap/docs/pull/21088/files#r2122709025

*   **Line 200**: In the example for `broker-node-3 ~/config/server.properties`, the `advertised.listeners` EXTERNAL value is `EXTERNAL://abc.eastus.azure.3199745.tidbcloud.com:9095`. Should this include the broker ID prefix like the others, e.g., `EXTERNAL://b3.abc.eastus.azure.3199745.tidbcloud.com:9095`? Please verify for consistency.

    ./kafka_2.13-3.7.1/bin/kafka-broker-api-versions.sh --bootstrap-server {one_of_broker_ip}:9092 | grep 9092
    # Expected output (the actual order might be different)
    {broker-node-1-ip}:9092 (id: 1 rack: null) -> (
    {broker-node-2-ip}:9092 (id: 2 rack: null) -> (
    {broker-node-3-ip}:9092 (id: 3 rack: null) -> (

    # Bootstrap from EXTERNAL listener
    ./kafka_2.13-3.7.1/bin/kafka-broker-api-versions.sh --bootstrap-server {one_of_broker_ip}:39092
    # Expected output for the last 3 lines (the actual order might be different)
    # The difference in the output from "bootstrap from INTERNAL listener" is that exceptions or errors might occur because advertised listeners cannot be resolved in kafka-pls-vnet.
    # TiDB Cloud will make these addresses resolvable and route requests to the correct broker when you create a changefeed connected to this Kafka cluster via Private Link Service.
    b1.abc.eastus.azure.3199745.tidbcloud.com:9093 (id: 1 rack: null) -> ERROR: org.apache.kafka.common.errors.DisconnectException
    b2.abc.eastus.azure.3199745.tidbcloud.com:9094 (id: 2 rack: null) -> ERROR: org.apache.kafka.common.errors.DisconnectException

>     b3.abc.eastus.azure.3199745.tidbcloud.com.com:9095 (id: 3 rack: null) -> ERROR: org.apache.kafka.common.errors.DisconnectException

Author: ```suggestion
    b3.abc.eastus.azure.3199745.tidbcloud.com.9095 (id: 3 rack: null) -> ERROR: org.apache.kafka.common.errors.DisconnectException
```

    - **Virtual network**: `kafka-pls-vnet`
    - **Subnet**: `brokers-subnet`
    - **Assignment**: `Dynamic`
    - **Availability zone**: `Zone-redundant`

4. In the **Backend pools** tab, add three backend pools as follows, and then click **Next : Inbound rules**.

    - Name: `pool1`; Backend Pool Configuration: `NIC`; IP configurations: `broker-node-1`
    - Name: `pool2`; Backend Pool Configuration: `NIC`; IP configurations: `broker-node-2`
    - Name: `pool3`; Backend Pool Configuration: `NIC`; IP configurations: `broker-node-3`

5. In the **Inbound rules** tab, add three load balancing rules as follows:

    1. Rule 1

        - **Name**: `rule1`
        - **IP version**: `IPv4`
        - **Frontend IP address**: `kafka-lb-ip`
        - **Backend pool**: `pool1`
        - **Protocol**: `TCP`
        - **Port**: `9093`
        - **Backend port**: `39092`
        - **Health probe**: click **Create New** and fill in the probe information.
            - **Name**: `kafka-lb-hp`
            - **Protocol**: `TCP`
            - **Port**: `39092`

    2. Rule 2

        - **Name**: `rule2`
        - **IP version**: `IPv4`
        - **Frontend IP address**: `kafka-lb-ip`

>         - **Backend pool**: `pool1`

Author: ```suggestion
        - **Backend pool**: `pool2`
```

        - **Protocol**: `TCP`
        - **Port**: `9094`
        - **Backend port**: `39092`
        - **Health probe**: click **Create New** and fill in the probe information.
            - **Name**: `kafka-lb-hp`
            - **Protocol**: `TCP`
            - **Port**: `39092`

    3. Rule 3

        - **Name**: `rule3`
        - **IP version**: `IPv4`
        - **Frontend IP address**: `kafka-lb-ip`

>         - **Backend pool**: `pool1`

Author: ```suggestion
        - **Backend pool**: `pool3`
```
